pyanaconda/modules/subscription/subscription_interface.py

```diff
@@ -0,0 +1,152 @@
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterface
+from pyanaconda.dbus.interface import dbus_interface
+
+
+@dbus_interface(SUBSCRIPTION.interface_name)
+class SubscriptionInterface(KickstartModuleInterface):
+    """DBus interface for Subscription module."""
+
+    def connect_signals(self):
+        super().connect_signals()
+        self.watch_property("Role", self.implementation.role_changed)
+        self.watch_property("SLA", self.implementation.sla_changed)
+        self.watch_property("Usage", self.implementation.usage_changed)
+        self.watch_property("Addons", self.implementation.addons_changed)
+
+        # the SystemPurposeWillBeSet property depends on the value of
+        # all other system purpose properties
+        self.watch_property("IsSystemPurposeSet", self.implementation.role_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.sla_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.usage_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.addons_changed)
+
+    @property
+    def ValidRoles(self) -> List[Str]:
+        """Return all valid roles."""
+        return self.implementation.valid_roles
+
+    @property
+    def Role(self) -> Str:
+        """Role for system subscription purposes."""
+        return self.implementation.role
+
+    @emits_properties_changed
+    def SetRole(self, role: Str):
+        """Set the intended role.
+
+        Sets the role intent for subscription purposes.
+
+        This setting is optional.
+
+        Example: Foo Role
+
+        :param str role: a role string
+        """
+        self.implementation.set_role(role)
+
+    @property
+    def ValidSLAs(self) -> List[Str]:
+        """Return all valid SLAs."""
+        return self.implementation.valid_slas
+
+    @property
+    def SLA(self) -> Str:
+        """SLA for system subscription purposes."""
+        return self.implementation.sla
+
+    @emits_properties_changed
+    def SetSLA(self, sla: Str):
+        """Set the intended SLA.
+
+        Sets the SLA intent for subscription purposes.
+
+        This setting is optional.
+
+        Example: Premium
+
+        :param str sla: a SLA string
+        """
+        self.implementation.set_sla(sla)
+
+    @property
+    def ValidUsageTypes(self) -> List[Str]:
+        """List all valid usage types."""
+        return self.implementation.valid_usage_types
+
+    @property
+    def Usage(self) -> Str:
+        """Usage for system subscription purposes."""
+        return self.implementation.usage
+
+    @emits_properties_changed
+    def SetUsage(self, usage: Str):
+        """Set the intended usage.
+
+        Sets the usage intent for subscription purposes.
+
+        This setting is optional.
+
+        Example: Production
```

Comment:
I don't see benefit of this example. If it should help please write multiple examples or none.

Same works for others.